model/metrics/dotnet-metrics.yaml

```diff
@@ -0,0 +1,213 @@
+groups:
+  - id: metric.dotnet.gc.collections
+    type: metric
+    metric_name: dotnet.gc.collections
+    brief: "Number of garbage collections that have occurred since the process has started."
+    note: "This metric reports the same values as calling [`GC.CollectionCount(int generation)`](https://learn.microsoft.com/dotnet/api/system.gc.collectioncount)."
+    instrument: counter
+    unit: "{collection}"
+    stability: experimental
+    attributes:
+      - ref: dotnet.gc.heap.generation
+        requirement_level: required
+
+  - id: metric.dotnet.gc.objects.size
```

Comment:
I was talking with Maoni (.NET GC architect) about this counter. We agreed that showing the change in heap size in between GCs is important, but also that this particular counter is confusing because it excludes fragmentation vs. all the other size counters which include fragmentation. We were thinking it would be preferable to remove this counter for now so that we can add a similar counter in .NET 10 that is the live heap size including fragmentation.

Some folks may complain about its absence but at least its easy to work-around in the short term by defining their own counter that wraps GC.GetTotalMemory().